src/main/kotlin/Main.kt

```diff
@@ -1,31 +1,23 @@
+import kotlinx.cli.*
 import kotlinx.cli.ArgParser
 import kotlinx.cli.ArgType
 import kotlinx.cli.required
 import kotlinx.coroutines.runBlocking
 
-
-fun main(args: Array<String>) = runBlocking {
-    val parser = ArgParser("mempool-cli")
-    val startHeight by parser.option(ArgType.Int, shortName = "s", description = "Start height").required()
-
+fun main(args: Array<String>) = runBlocking() {
```

Comment:
Same here, you should avoid `try/catch`, there are better ways of handling it in kotlin:

```kotlin
fun main(args: Array<String>) = runBlocking {
    val parser = ArgParser("kotlin-cli")
    val startHeight by parser.option(ArgType.Int, shortName = "s", description = "Start Height").required()

    parser.parse(args)

    val apiClient = MempoolClient()

    apiClient.fetchFirstBlockId(startHeight).fold(
        onSuccess = { blockId ->
            apiClient.fetchTransactionIds(blockId).fold(
                onSuccess = { txIds -> txIds.forEach { println(it) } },
                onFailure = { e -> println("error fetching transaction IDs: $e") }
            )
        },
        onFailure = { e -> println("error fetching block ID: $e") }
    )
}
```